[GEOT-5726] Parse an empty CSSParameter as an empty string instead of null

tbarsballe opened the pull request:
See https://osgeo-org.atlassian.net/browse/GEOT-5726

jodygarnett commented on modules/library/main/src/main/java/org/geotools/styling/SLDParser.java:
```diff
@@ -2308,6 +2308,9 @@ private Expression parseCssParameter(Node root, boolean trimWhiteSpace) {
             ret = manageMixed(ret, expression);
         }
+        if (ret == null) {
+            return ff.literal("");
```
I would prefer to use null for something that is not there?

jodygarnett replied:
Double check the GTXML parser handles an empty tag in the same way or we will get into inconsistencies in our model.

jodygarnett commented on modules/library/main/src/test/java/org/geotools/styling/SLDTransformerTest.java:
```diff
@@ -834,6 +830,50 @@ public void testTextSymbolizerTransformOutAndInAndOutAgain() throws Exception {
+
+        assertNotNull(firstImport[0]);
+
+        // NPE here
```
there is no longer a NPE exception here

jodygarnett commented on modules/library/main/src/test/java/org/geotools/styling/SLDTransformerTest.java:
```diff
@@ -834,6 +830,50 @@ public void testTextSymbolizerTransformOutAndInAndOutAgain() throws Exception {
     }
+
+    @Test
+    public void testTextSymbolizerTransformOutAndInAndOutAndInAgainWithEmptyFontFamily() throws Exception {
```
for a test case this specific quote the bug number or people will think you are nuts

tbarsballe commented:
Addressed feedback - added / improved comments.

I added a test case for consistency with the GTXML parser. Turns out it doesn't match the either the current behavior (getting `null`) or the new behavior (getting `""`). Instead, it fills in the default, in this case `"Serif"`. I've added this test case, but set it as ignored, in case we revisit this behavior in the future

tbarsballe commented:
Backported to 17.x
